### PR TITLE
[12.0][UPD] openupgrade_records: add model_type for model records

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Applying https://github.com/OCA/OpenUpgrade/pull/2004#discussion_r323733085 comment.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Can you apply it to analysis files?

**Comment by MiquelRForgeFlow** [comment]
I will do it later, after a new update upstream.

**Comment by pedrobaeza** [comment]
Yeah, but without it, it's difficult to analyze if it's correct or not.

File: odoo/addons/openupgrade_records/models/openupgrade_record.py [hunk 1]
```diff
@@ -41,17 +41,37 @@ class Record(models.Model):
...
+    def _compute_model_type(self):
+        for rec in self:
+            if rec.type == 'model':
+                model = self.env[rec.model]
+                if model._auto and model._transient:
+                    rec.model_type = 'transient'
+                elif model._auto:
+                    rec.model_type = ''
+                else:
+                    rec.model_type = 'abstract'
```
Comment (pedrobaeza) [inline odoo/addons/openupgrade_records/models/openupgrade_record.py:72]: `_auto = False` is not the same as `abstract`, so you must distinguish both cases

Comment (MiquelRForgeFlow) [reply odoo/addons/openupgrade_records/models/openupgrade_record.py:72]: Well, and how I call those models that are `_auto=False` but are not defined as `abstract`?

Comment (pedrobaeza) [reply odoo/addons/openupgrade_records/models/openupgrade_record.py:72]: I would bet for "sql view"

Comment (MiquelRForgeFlow) [reply odoo/addons/openupgrade_records/models/openupgrade_record.py:72]: done

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**